nixos/modules/services/web-servers/nginx/default.nix

```
@@ -318,15 +318,15 @@ let
 
         listenString = { addr, port, ssl, extraParameters ? [], ... }:
           # UDP listener for QUIC transport protocol.
-          (if ssl && vhost.quic then "
+          (optionalString (ssl && vhost.quic) "
             listen ${addr}:${toString port} quic "
           + optionalString vhost.default "default_server "
           + optionalString vhost.reuseport "reuseport "
           + optionalString (extraParameters != []) (concatStringsSep " " (
             let inCompatibleParameters = [ "ssl" "proxy_protocol" "http2" ];
                 isCompatibleParameter = param: !(any (p: p == param) inCompatibleParameters);
             in filter isCompatibleParameter extraParameters))
-          + ";" else "")
+          + ";")
```

```suggestion
          (optionalString (ssl && vhost.quic) ("
            listen ${addr}:${toString port} quic "
          + optionalString vhost.default "default_server "
          + optionalString vhost.reuseport "reuseport "
          + optionalString (extraParameters != []) (concatStringsSep " " (
            let inCompatibleParameters = [ "ssl" "proxy_protocol" "http2" ];
                isCompatibleParameter = param: !(any (p: p == param) inCompatibleParameters);
            in filter isCompatibleParameter extraParameters))
          + ";" else "")
          + ";"))
```
Can someone test if this fixes it? I am lacking the time today.

---

Already applied in e25dc4a95ed69f37ce443b8fcad00fb9337e6eed